images/macos/provision/core/codeql-bundle.sh

```diff
@@ -2,24 +2,41 @@
 source ~/utils/utils.sh
 
 # Retrieve the name of the CodeQL bundle preferred by the Action (in the format codeql-bundle-YYYYMMDD).
-codeql_bundle_name="$(curl -sSL https://raw.githubusercontent.com/github/codeql-action/v2/src/defaults.json | jq -r .bundleVersion)"
-# Convert the bundle name to a version number (0.0.0-YYYYMMDD).
-codeql_bundle_version="0.0.0-${codeql_bundle_name##*-}"
+base_url="$(curl -sSL https://raw.githubusercontent.com/github/codeql-action/v2/src/defaults.json)"
+codeql_tag_name="$(echo "$base_url" | jq -r '.bundleVersion')"
+codeql_cli_version="$(echo "$base_url" | jq -r '.cliVersion')"
+prior_codeql_tag_name="$(echo "$base_url" | jq -r '.priorBundleVersion')"
+prior_codeql_cli_version="$(echo "$base_url" | jq -r '.priorCliVersion')"
 
-echo "Downloading CodeQL bundle $codeql_bundle_version..."
-download_with_retries "https://github.com/github/codeql-action/releases/download/$codeql_bundle_name/codeql-bundle.tar.gz" "/tmp" "codeql-bundle.tar.gz"
-codeqlArchive="/tmp/codeql-bundle.tar.gz"
+# Convert the tag names to bundles with a version number (x.y.z-YYYYMMDD).
+codeql_bundle_version="${codeql_cli_version}-${codeql_tag_name##*-}"
+prior_codeql_bundle_version="${prior_codeql_cli_version}-${prior_codeql_tag_name##*-}"
 
-codeqlToolcachePath="$AGENT_TOOLSDIRECTORY/codeql/$codeql_bundle_version/x64"
-mkdir -p $codeqlToolcachePath
+# Download and name both CodeQL bundles.
+codeql_bundle_versions=("${codeql_bundle_version}" "${prior_codeql_bundle_version}")
+codeql_tag_names=("${codeql_tag_name}" "${prior_codeql_tag_name}")
 
-echo "Unpacking the downloaded CodeQL bundle archive..."
-tar -xzf $codeqlArchive -C $codeqlToolcachePath
+for ((i=0;i<=1;i++)); do
```

Comment:
Could be something like `for $version in $codeql_bundle_versions[@]; do ... done`